### DISCUSSION TITLE
Getting metadata variable from server action

### DISCUSSION THREAD

**Discussion by laurent512** [next-safe-action/next-safe-action#236] [General]
Hello, 

I defined the following 

```
 export const updateUserAction = customActionClient
    .metadata({
        action: "updateUser",
    })
    .....
```
For some RBAC logic between front end and backend, I would like to retrieve the action field from the action. 
Is there a possibility to do it ? something like updateUserAction.metadata.action ? 

Thank you so much :)

**Comment by TheEdoRan** [comment]
Hi, the problem is that `updateUserAction` is the Server Action that gets called by the Client Component, so the return values are available to the client, by design. `metadata` is server-side information, and I don't think it's a good idea to expose it to the client. A possible solution would be to return it inside your action, so it's available via `result.data.metadata`.
